src/CheckoutSdk/Payments/CustomerSource.cs

```diff
@@ -0,0 +1,36 @@
+﻿using System;
+
+namespace Checkout.Sdk.Payments
+{
+    public class CustomerSource : IPaymentSource
+    {
+        public CustomerSource(string id, string email)
+        {
+            if (string.IsNullOrWhiteSpace(id)
+                && string.IsNullOrWhiteSpace(email))
+            {
+                throw new CheckoutException($"Either {id} or {email} must be provided.");
```

Comment:
I would reserve using CheckoutException for SDK specific functionality. For argument checking, the standard exception types (ArguemntException, InvalidOperationException etc.) should suffice